### ISSUE TITLE
[Request] Proper background/detached mode.

### ISSUE THREAD

**Issue by skyzyx** [iann0036/iamlive#20]
By default, this tool runs in the foreground and prints data to stdout.

It would be nice to enable a _proper_ background/detached mode, which:

1. Runs in the background.
2. When run in the background, it returns its PID.
3. Implies the use of `-output-file`.
4. Background mode would not write to stdout.

That would let me do something like this:

```bash
ACCOUNT_ID=$(aws sts get-caller-identity | jq -Mr '.AccountId')

IAMLIVE_PID=$(
    iamlive \
        --detached \
        --mode proxy \
        --output-file policy.json \
        --refresh-rate 1 \
        --sort-alphabetical \
        --bind-addr 127.0.0.1:10080 \
        --ca-bundle ~/.iamlive/ca.pem \
        --ca-key ~/.iamlive/ca.key \
        --account-id "$ACCOUNT_ID"
)

AWS_CSM_ENABLED=true
HTTP_PROXY=http://127.0.0.1:10080
HTTPS_PROXY=http://127.0.0.1:10080
AWS_CA_BUNDLE=~/.iamlive/ca.pem

terraform apply -auto-approve
terraform destroy -auto-approve

kill "$IAMLIVE_PID"
```

It also means that `policy.json` would need to be written even if a `SIGKILL` is received. (If there's a different signal that should be sent, documenting it would be valuable.)

**Comment by iann0036** [comment]
Implemented as `--background`, available in v0.19.

Will write and quit if receiving a SIGINT, SIGTERM (your example) or SIGQUIT.

Will write and continue running if receiving a SIGHUP.

**Comment by skyzyx** [comment]
👍🏻
